### PR TITLE
Additional summary stats using timeline

### PR DESCRIPTION
**Pull request by AlexTheLion123**
Adds some more optional statistics to calculate_summary_statistics(). Corresponding with @tsorro  to align answers for the statistics for opening for review.

### CODE REVIEW

**Comment by AlexTheLion123** [comment]
Fixes #93

**Comment by AlexTheLion123** [comment]
Changing `average duration of winning/losing trades` to use number of candles/bars. A “day” is too long a unit of measure.

**Comment by AlexTheLion123** [comment]
> TODO: Create new ticket for changing `average duration of winning/losing trades` to use number of candles/bars. A “day” is too long a unit of measure.

Have created #96 for this

**Comment by AlexTheLion123** [comment]
With this PR, the `calculate_summary_statistics` method now has a total of 3 optional key-word only arguments. It is backwards compatible so no arguments have to be provided. 

```python
from IPython.core.display_functions import display
from tradeexecutor.analysis.trade_analyser import expand_timeline_raw

timeline = analysis.create_timeline()

raw_timeline = expand_timeline_raw(
        universe.universe.exchanges,
        universe.universe.pairs,
        timeline,
        initial_deposit)

summary = analysis.calculate_summary_statistics(raw_timeline=raw_timeline, stop_loss_pct=0, time_bucket=candle_time_bucket)

with pd.option_context("display.max_row", None):
    display(summary.to_dataframe())
```

File: tradeexecutor/analysis/trade_analyser.py [hunk 1]
```diff
@@ -457,8 +482,29 @@ def get_open_positions(self) -> Iterable[Tuple[PrimaryKey, TradePosition]]:
...
+            :param stop_loss_pct:
+            stop loss percentage
+
+            :param time_bucket:
+            time bucket to display average duration as 'number of bars' instead of 'number of days'. 
+        """
+
+        if(raw_timeline is not None):
+            assert isinstance(raw_timeline, pd.DataFrame), "Not a valid timeline, use expand_timeline_raw() method to create it"
+        if(stop_loss_pct is not None):
```
Comment (miohtama) [inline tradeexecutor/analysis/trade_analyser.py:503]: Stop loss percent cannot be given as a parameter here. It is part of `TradingPosition` instance. 

- Each position can have its own stop loss. 
- Stop loss must be passed around other position parameters when the timeline is created

Comment (AlexTheLion123) [reply tradeexecutor/analysis/trade_analyser.py:503]: According to @tsorro, the `max capital risk at stop loss` is calculated as follows:

How much of the total capital do we risk with the calculated stop loss.  

$$\frac{Current stop loss percentage * the position size}{Current total capital in the strategy} * 100$$

Comment (miohtama) [reply tradeexecutor/analysis/trade_analyser.py:503]: You can get all this information as the part of the `Portfolio` object. 

- Total capital in the strategy: `portfolio.get_total_equity()` - https://tradingstrategy.ai/docs/programming/api/execution/help/tradeexecutor.state.portfolio.Portfolio.html#tradeexecutor.state.portfolio.Portfolio.get_total_equity
- Position size: For position we need to add a new method `position.get_max_value()` that is the maximum value of the position over its lifetime. The position value can move when additional trades are made. (ticket 1)
- - Note that the total capital of the strategy is based on the current (at the end of the backtest value) - we would need to add a new position variable `position.portfolio_value_at_open` so we can calculate the risk at the time of the opening the position (ticket 2)
- Then when the timeline data is exported - we have the access these functions and export them part of the analysis (ticket 3)

**Pull request merged**